### PR TITLE
Use custom path mapping supported by iRODS CSI Driver v0.3.0

### PR DESCRIPTION
**Pull request by iychoi**
Use custom path mapping supported by iRODS CSI Driver v0.3.0
This allows mounting data objects and collections at different paths on a single mount path.
The new code will mount both input and output of a job on "/data".
Input data objects and collections are located at "/data/input".
Output directory is located at "/data/output".

### CODE REVIEW

**Comment by cyverse-org** [comment]
Can one of the admins verify this patch?

File: internal/volumes.go [hunk 1]
```diff
@@ -15,278 +16,157 @@ var (
...
+				resourceType := "file"
+				if stepInput.Type == "FileInput" || strings.ToLower(stepInput.Type) == "fileinput" {
+					resourceType = "file"
+				} else if stepInput.Type == "MultiFileSelector" || strings.ToLower(stepInput.Type) == "multifileselector" {
+					resourceType = "file"
+				} else if stepInput.Type == "FolderInput" || strings.ToLower(stepInput.Type) == "folderinput" {
+					resourceType = "dir"
+				} else {
+					// unknown
+					return nil, fmt.Errorf("unknown step input type - %s", stepInput.Type)
```
Comment (slr71) [inline internal/volumes.go:58]: For cases where the first condition is matched by a subset of values matched by the second condition, I think I'd omit the first condition:

```suggestion
				resourceType := "file"
				if strings.ToLower(stepInput.Type) == "fileinput" {
					resourceType = "file"
				} else if strings.ToLower(stepInput.Type) == "multifileselector" {
					resourceType = "file"
				} else if strings.ToLower(stepInput.Type) == "folderinput" {
					resourceType = "dir"
				} else {
					// unknown
					return nil, fmt.Errorf("unknown step input type - %s", stepInput.Type)
```

**Review by slr71 — APPROVED** [review]
Thanks for working on this! I had one minor suggestion, but I didn't spot any problems in the code changes that you made.

I do have one concern about the command line that is being built. It appears that the way that command line is being built doesn't change depending on whether or not the CSI driver is being used, and it probably should change if any input parameters are present on the command line. Most VICE apps don't actually specify input parameters on the command line, so I don't think that that has to change for this PR. I will create a Jira issue just to remind myself that it needs to be addressed before we enable the CSI driver in production, though.

I'm also unable to test these changes for the time being because the QA environment is partially down at the moment. I'll make a note to do some testing next week.

**Comment by slr71** [comment]
Here's the link to the Jira issue that I mentioned above for reference: https://cyverse.atlassian.net/browse/CORE-1345

File: internal/volumes.go (hunk 1 above)
Comment (iychoi) [reply internal/volumes.go]: Applied.

**Comment by iychoi** [comment]
Updated iRODS FUSE Lite path mapping configuration to create collections if not exists. This is required for output collection.

**Comment by slr71** [comment]
Merging this now.

**Pull request merged**